pkg/standalone-utils/test/VaultActions.test.ts

```diff
@@ -1161,4 +1190,324 @@ describe('VaultActions', function () {
       });
     });
   });
+
+  describe('user balance ops', () => {
+    const amountDAI = fp(2);
+    const amountSNX = fp(5);
+
+    context('when caller is not authorized', () => {
+      it('reverts', async () => {
+        await expect(
+          relayer.connect(other).multicall([
+            encodeManageUserBalance({
+              ops: [
+                {
+                  kind: UserBalanceOpKind.DepositInternal,
+                  asset: tokens.DAI.address,
+                  amount: amountDAI,
+                  sender: user.address,
+                },
+              ],
+            }),
+          ])
+        ).to.be.revertedWith('Incorrect sender');
+      });
+    });
+
+    context('when caller is authorized', () => {
+      let sender: Account;
+
+      context('sender = user', () => {
+        beforeEach(() => {
+          sender = user;
+        });
+
+        itTestsUserBalance();
+      });
+
+      context('sender = relayer', () => {
+        sharedBeforeEach('fund relayer with tokens and approve vault', async () => {
+          sender = relayer;
+          await tokens.DAI.transfer(relayer, amountDAI, { from: user });
+          await tokens.SNX.transfer(relayer, amountSNX, { from: user });
+          await approveVaultForRelayer(relayerLibrary, user, tokens);
+        });
+
+        itTestsUserBalance();
+      });
+
+      function itTestsUserBalance() {
+        it('sends immediate amounts', async () => {
+          // Internal balance of sender doesn't change
+          // Tokens are transferred from sender to recipient's internal balance
+
+          await expectBalanceChange(
```

Comment:
```suggestion
          // Internal balance of sender doesn't change
          // Tokens are transferred from sender to recipient's internal balance
          // `expectBalanceChange` in this case will track internal balances, since we are passing the vault as an argument.
          await expectBalanceChange(
```

Reply:
Yes; that's confusing/unintuitive behavior of `expectBalanceChange`. I wanted to keep the tests as similar as possible, so I didn't change anything with it, but I agree it could benefit from a comment here. I'll probably expand even more.